src/lib.rs

```diff
@@ -127,9 +177,24 @@ impl Collector {
     /// # use metrics_process::Collector;
     /// let collector = Collector::default().prefix("my_prefix_");
     /// ```
-    pub fn prefix(mut self, prefix: impl Into<String>) -> Self {
-        self.prefix = prefix.into();
-        self
+    pub fn prefix(self, prefix: impl Into<String>) -> Self {
```

Comment:
Please add `deprecated` attribute (with `since="1.1.0") and explain it in comments in favor of new `Collector::new()` function.